File: src/main/java/hazae41/localtab/Lang.java
package hazae41.localtab;

class Langs {
  static Lang getLang(String locale) {
    String first = locale.split("_")[0];

    if (first.equals("en"))
      return new English();
    if (first.equals("fr"))
      return new French();
    if (first.equals("ca"))
      return new Spanish();
    if (first.equals("es"))
      return new Spanish();
    if (first.equals("pt"))
      return new Portuguese();
    if (first.equals("it"))
      return new Italian();
    if (first.equals("de"))
      return new German();
    if (first.equals("nl"))
      return new Dutch();
    if (first.equals("cs"))
      return new Czech();
    if (first.equals("ru"))
      return new Russian();
    if (first.equals("pl"))
      return new Polish();
    if (first.equals("ja"))
      return new Japanese();
    if (first.equals("ko"))
      return new Korean();
    if (first.equals("zh"))
      return new Chinese();

    return new English();
  }
}

public abstract class Lang {
  abstract String getHeader();
}

class English extends Lang {
  @Override
  String getHeader() {
    return "Close players";
  }
}

class French extends Lang {
  @Override
  String getHeader() {
    return "Joueurs proches";
  }
}

class Spanish extends Lang {
  @Override
  String getHeader() {
    return "Jugadores cercanos";
  }
}

class Portuguese extends Lang {
  @Override
  String getHeader() {
    return "Jogadores próximos";
  }
}

class Italian extends Lang {
  @Override
  String getHeader() {
    return "Giocatori vicini";
  }
}

class German extends Lang {
  @Override
  String getHeader() {
    return "Enge Spieler";
  }
}

class Dutch extends Lang {
  @Override
  String getHeader() {
    return "Nauwe spelers";
  }
}

class Czech extends Lang {
  @Override
  String getHeader() {
    return "Blízcí hráči";
  }
}

class Russian extends Lang {
  @Override
  String getHeader() {
    return "Близкие игроки";
  }
}

class Polish extends Lang {
  @Override
  String getHeader() {
    return "Bliscy gracze";
  }
}

class Chinese extends Lang {
  @Override
  String getHeader() {
    return "亲密玩家";
  }
}

class Japanese extends Lang {
  @Override
  String getHeader() {
    return "近いプレーヤー";
  }
}

class Korean extends Lang {
  @Override
  String getHeader() {
    return "가까운 플레이어";
  }
}

File: src/main/kotlin/Lang.kt
package hazae41.localtab

fun getLang(locale: String): Lang {
  val first = locale.split("_")[0]

  if (first == "en")
    return English
  if (first == "fr")
    return French
  if (first == "es")
    return Spanish
  if (first == "ca")
    return Spanish
  if (first == "it")
    return Italian
  if (first == "de")
    return German
  if (first == "nl")
    return Dutch
  if (first == "cs")
    return Czech
  if (first == "ru")
    return Russian
  if (first == "pl")
    return Polish
  if (first == "zh")
    return Chinese
  if (first == "ja")
    return Japanese
  if (first == "ko")
    return Korean

  return English
}

interface Lang {
  fun header(): String
}

object English : Lang {
  override fun header() = "Close players"
}

object French : Lang {
  override fun header() = "Joueurs proches"
}

object Spanish : Lang {
  override fun header() = "Jugadores cercanos"
}

object Italian : Lang {
  override fun header() = "Giocatori vicini"
}

object German : Lang {
  override fun header() = "Enge Spieler"
}

object Dutch : Lang {
  override fun header() = "Nauwe spelers"
}

object Czech : Lang {
  override fun header() = "Blízcí hráči"
}

object Russian : Lang {
  override fun header() = "Близкие игроки"
}

object Polish : Lang {
  override fun header() = "Bliscy gracze"
}

object Chinese : Lang {
  override fun header() = "亲密玩家"
}

object Japanese : Lang {
  override fun header() = "近いプレーヤー"
}

object Korean : Lang {
  override fun header() = "가까운 플레이어"
}